### PR TITLE
UX-373 Upgrade styled-components to 5.2

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Upgrades to `styled-components` v5.2
- Upgrades to `jest-styled-components` v7
- Replaces package peer-dependencies with `styled-components` v5.1.0.
- Globally installs `jest-styled-components` in jest's setup file instead of in every test file
- Modifies some tests that were failing from the dependency upgrades
  - Needed to scope some `wrappers` within arrow function
  - Needed to specify `.at(x)` in a few places that were detecting multiple DOM nodes
  - Explicitly added a button in ListBox stories for the Cypress focus tests
- Adds a `clean` helper to filter out styling props
- Updates `react-syntax-highlighter` for the CodeBlock stories

### PR Checklist

- [x] Add the correct `type` label
- [x] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: config/jest/setup.js [hunk 1]
```diff
@@ -2,8 +2,8 @@
 import './raf';
 import Enzyme from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
-import toJson from 'enzyme-to-json';
-import * as matchers from './matchers'
+import * as matchers from './matchers';
+import 'jest-styled-components';
```
Comment (jonambas) [inline config/jest/setup.js:6]: Moved `jest-styled-component` to a global install here

File: babel.config.js [hunk 2]
```diff
@@ -12,7 +12,10 @@ module.exports = {
         '@babel/preset-react',
         'react-app',
       ],
-      plugins: ['@babel/plugin-proposal-class-properties'],
+      plugins: [
+        '@babel/plugin-proposal-class-properties',
+        ['babel-plugin-styled-components', { ssr: true, displayName: false, namespace: 'sc-' }],
```
Comment (jonambas) [inline babel.config.js:17]: This plugin is required to run `jest-styled-components` v7 with `styled-components` v5

File: jest.config.json [hunk 3]
```diff
@@ -30,6 +30,7 @@
     "^react-native$": "react-native-web",
     "\\.(css|scss)$": "identity-obj-proxy",
     "@sparkpost/matchbox-icons(.*)": "<rootDir>/packages/matchbox-icons/src",
-    "@sparkpost/matchbox(.*)": "<rootDir>/packages/matchbox/src"
+    "@sparkpost/matchbox(.*)": "<rootDir>/packages/matchbox/src",
+    "^styled-components": "<rootDir>/node_modules/styled-components"
```
Comment (jonambas) [inline jest.config.json:34]: Resolving one styled-components package in tests

File: packages/matchbox/src/components/Banner/Banner.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { deprecate } from '../../helpers/propTypes';
-import { omit } from '../../helpers/systemProps';
+import { omit } from '../../helpers/props';
```
Comment (jonambas) [inline packages/matchbox/src/components/Banner/Banner.js:4]: Changed this helper file name

File: packages/matchbox/src/helpers/props.js [hunk 5]
```diff
@@ -41,3 +41,17 @@ export function pick(props, names) {
...
 }
+
+/**
+ * Returns a configuration object for styled-components with omitted props, to be used with .withConfig
+ * @param {Array} arr
+ * @param {Object} config
+ *
+ * @see https://styled-components.com/docs/api#shouldforwardprop
+ */
+export function clean(arr = [], config = {}) {
```
Comment (jonambas) [inline packages/matchbox/src/helpers/props.js:52]: Here is the new helper

File: stories/form/ListBox.stories.js [hunk 6]
```diff
@@ -21,12 +21,16 @@ export const BasicListbox = withInfo()(() => (
...
-  </ListBox>
+  <>
+    <ListBox id="listbox-1" defaultValue="bravo" label="Select an option">
+      <ListBox.Option value="alpha">Alpha</ListBox.Option>
+      <ListBox.Option value="bravo">Bravo</ListBox.Option>
+      <ListBox.Option value="charlie">Charlie</ListBox.Option>
+      <ListBox.Option value="delta">Delta</ListBox.Option>
+    </ListBox>
+    {/* This is here for cypress tests */}
+    <button>end focus test</button>
```
Comment (jonambas) [inline stories/form/ListBox.stories.js:32]: Added an explicit button here for the Cypress focus tests

File: packages/matchbox/src/components/Panel/legacy/tests/Panel.test.js [hunk 7]
```diff
@@ -1,25 +1,25 @@
...
+  const subject = () =>
+    global.mountStyled(
+      <Panel accent sectioned title="test title">
+        Panel Content
+      </Panel>,
+    );
 
   it('renders panel styles', () => {
-    expect(wrapper).toHaveStyleRule('background', tokens.color_white);
+    expect(subject()).toExist('background', tokens.color_white);
```
Comment (jonambas) [inline packages/matchbox/src/components/Panel/legacy/tests/Panel.test.js:14]: Oops need to switch this assertion back

File: packages/matchbox/src/helpers/props.js (hunk 5 above)
Comment (logansparlin) [reply packages/matchbox/src/helpers/props.js:52]: This looks great

**Review by logansparlin — APPROVED** [review]
This looks great! Just one small comment about a package version but good to merge if you're good with that

File: package.json [hunk 8]
```diff
@@ -76,9 +78,10 @@
     "@testing-library/cypress": "^5.3.1",
     "autoprefixer": "^8.4.1",
     "babel-eslint": "^10.0.3",
-    "babel-jest": "^24.9.0",
+    "babel-jest": "^26.6.3",
     "babel-loader": "^8.0.6",
     "babel-plugin-istanbul": "^6.0.0",
+    "babel-plugin-styled-components": "^1.11.1",
     "babel-preset-es2015": "^6.24.1",
     "babel-preset-react": "^6.24.1",
```
Comment (jonambas) [inline package.json:86]: Can maybe uninstall

File: packages/matchbox/src/components/Popover/tests/Popover.test.js [hunk 9]
```diff
@@ -63,50 +62,59 @@ describe('Popover', () => {
   describe('content', () => {
     it('should render deprecated sectioned prop', () => {
       const wrapper = subject({ sectioned: true, open: true });
-      expect(wrapper.find('[data-id="popover-content"]')).toHaveStyleRule('padding', '1rem');
+      expect(wrapper.find('[data-id="popover-content"]').first()).toHaveStyleRule(
```
Comment (jonambas) [inline packages/matchbox/src/components/Popover/tests/Popover.test.js:65]: Use `.at(0)`

**Review by logansparlin — APPROVED** [review]
Looks good

**Pull request merged**